src/tests/sysdb-tests.c

```diff
@@ -1827,6 +1844,7 @@ START_TEST (test_sysdb_remove_nonexistent_group)
 }
 END_TEST
 
+#ifdef BUILD_LOCAL_PROVIDER
```

Comment:
OTOH, I think this ifdef is fine because the get_new_id function is only useful for the local provider.